### DISCUSSION TITLE
Guide for Local development with Vercel Postgres

### DISCUSSION THREAD

**Discussion by mmikhan** [vercel/next.js#71534] [Ideas]
### Goals

1. An official guide on Next.js documentation that shows how to setup a local Postgres with Vercel Postgres using Docker

### Background

Setting up a local development environment is the first thing that everyone does. It's very surprising that Next.js site has no official documentation on how to setup a local Postgres with Docker. The only [documentation I could found is on Vercel](https://vercel.com/docs/storage/vercel-postgres/local-development) but that doesn't work either.

Not sure whether this will get priority from the Next.js team anytime soon but this will makes all the devs life easier. Especially any new comer coming to Next.js

### Proposal

Write a clear documentation to run a local development environment with Vercel Postgres through Docker

**Comment by leerob** [comment]
We have recently changed from having a single Postgres provider through Vercel, to instead having **many different Postgres options** (or other databases) through the [Vercel Marketplace](https://vercel.com/marketplace). This means that the local development setup will be provider-dependent.

For example:

- https://neon.tech/guides/local-development-with-neon
- https://supabase.com/docs/guides/local-development/overview
- And so on, for each provider.

Another option is one of the official Next.js starter templates, [like this SaaS starter](https://github.com/nextjs/saas-starter), which includes a CLI that can set up a local Postgres instance through Docker. Then, since the template uses Drizzle as the ORM, you can use that for database migrations and viewing your data through a GUI (Drizzle Studio).
